category_encoders/utils.py

```diff
@@ -175,7 +184,281 @@ def get_generated_cols(X_original, X_transformed, to_transform):
     return current_cols
 
 
+class EncodingRelation(Enum):
```

Comment:
I introduced it in the first place to add the output shape of the transform method dynamically in the docstring, but didn't get it to work since docstrings are not meant to be dynamic (I guess).  
I then decided to keep it since it could be useful metadata (e.g. you could add a test that the expected number of output columns is generated)

Reply:
which could be actually a useful test by the way since when working on this I discovered that there might be problems with e.g. the backward difference encoder. This adds some intercept column and I guess if you encode two columns the latter intercept column overwrites the first one. But I'll have to check. If this turns out to be the case I'll create a separate issue for it